python_src/src/lamp_py/tableau/hyper.py

```diff
@@ -0,0 +1,348 @@
+        if map_check is not None:
+            return map_check
+
+        if dtype.startswith("date32"):
+            return SqlType.date()
+
+        if dtype.startswith("timestamp"):
+            return SqlType.timestamp()
+
+        return SqlType.text()
+
+    def upload_parquet(self) -> None:
+        """
+        Upload local parquet file to remote path
+        """
+        process_logger = ProcessLogger(
+            "hyper_job_upload_parquet",
+            source=self.local_parquet_path,
+            destination=self.remote_parquet_path,
+        )
+        process_logger.log_start()
+        fs.copy_files(
+            source=self.local_parquet_path,
+            destination=self.remote_parquet_path,
+            source_filesystem=fs.LocalFileSystem(),
+            destination_filesystem=self.remote_fs,
+        )
+        process_logger.log_complete()
+
+    def download_parquet(self) -> None:
+        """
+        Download remote parquet file to local path
+        """
+        process_logger = ProcessLogger(
+            "hyper_job_download_parquet",
+            source=self.remote_parquet_path,
+            destination=self.local_parquet_path,
+        )
+        process_logger.log_start()
+
+        if os.path.exists(self.local_parquet_path):
+            os.remove(self.local_parquet_path)
+
+        fs.copy_files(
+            source=self.remote_parquet_path,
+            destination=self.local_parquet_path,
+            source_filesystem=self.remote_fs,
+            destination_filesystem=fs.LocalFileSystem(),
+        )
+
+        process_logger.log_complete()
+
+    def max_stats_of_parquet(self) -> Dict[str, str]:
+        """
+        Create dictionary of maximum value for each column of locally saved
+        parquet file
+        """
+        # get row_groups from parquet metadata (list of dicts)
+        row_groups = pq.read_metadata(self.local_parquet_path).to_dict()[
+            "row_groups"
+        ]
+
+        # explode columns element from all row  groups into flat list
+        parquet_column_stats = list(
+            chain.from_iterable(
+                [row_group["columns"] for row_group in row_groups]
+            )
+        )
+
+        return {
+            col["path_in_schema"]: col["statistics"].get("max")
+            for col in parquet_column_stats
+            if col["statistics"]
+        }
+
+    def create_local_hyper(self) -> int:
+        """
+        Create local hyper file, from remote parquet file
+        """
+        if os.path.exists(self.local_hyper_path):
+            os.remove(self.local_hyper_path)
+
+        # create HyperFile TableDefinition based on Job parquet schema
+        hyper_table_schema = TableDefinition(
+            table_name=self.hyper_table_name,
+            columns=[
+                TableDefinition.Column(
+                    col.name, self.convert_parquet_dtype(col.type)
+                )
+                for col in self.parquet_schema
+            ],
+        )
```

Comment:
Construct a HyperFile `TableDefinition`, two parameters are required, the `table_name` and `columns`. `columns` is a list of `TableDefinition.Column` objects.